test/spec/modules/rubiconBidAdapter_spec.js

```diff
@@ -1909,8 +1926,8 @@ describe('the rubicon adapter', function () {
           );
 
           const [request] = spec.buildRequests(bidderRequest.bids, bidderRequest);
-          expect(request.data.imp[0].ext.context.data.adserver.adslot).to.equal('1234567890');
-          expect(request.data.imp[0].ext.context.data.adserver.name).to.equal('adServerName1');
+          expect(request.data.site.ext.data.adserver.adslot).to.equal('1234567890');
+          expect(request.data.site.ext.data.adserver.name).to.equal('adServerName1');
```

Comment:
I do not think this is right.

PBS expects it in the `imp[].ext.context.data.adserver.adslot`

Because if you have multiple imps in a PBS request, you cannot just have a single adSlot in the top level site object.

I think the logic for adServer / slotName / pbAdSlot stuff needs to be outside of the applyFPD so it can be placed in the right spots. (Or you need to refference the imp[0].ext.context.data.adserver.adslot inside the function most likely.

Reply:
Moved back to imp[0] as opposed to top level